pymc3/distributions/continuous.py

```diff
@@ -2366,80 +2308,62 @@ class HalfCauchy(PositiveContinuous):
     beta: float
         Scale parameter (beta > 0).
     """
+    rv_op = halfcauchy
 
-    def __init__(self, beta, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.mode = aet.as_tensor_variable(0)
-        self.median = self.beta = aet.as_tensor_variable(floatX(beta))
+    @classmethod
+    def dist(cls, beta, *args, **kwargs):
+        beta = aet.as_tensor_variable(floatX(beta))
 
-        assert_negative_support(beta, "beta", "HalfCauchy")
+        # mode = aet.as_tensor_variable(0)
+        # median = beta
 
-    def _random(self, beta, size=None):
-        u = np.random.uniform(size=size)
-        return beta * np.abs(np.tan(np.pi * (u - 0.5)))
-
-    def random(self, point=None, size=None):
-        """
-        Draw random values from HalfCauchy distribution.
+        assert_negative_support(beta, "beta", "HalfCauchy")
+        return super().dist([beta], **kwargs)
 
-        Parameters
-        ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
 
-        Returns
-        -------
-        array
-        """
-        # beta = draw_values([self.beta], point=point, size=size)[0]
-        # return generate_samples(self._random, beta, dist_shape=self.shape, size=size)
+@_logp.register(HalfCauchyRV)
+def half_cauchy_logp(op, value, beta, alpha):
```

Comment:
I'm a bit sceptical as for why this function over here requires the extra `alpha` input argument even though it does not use it. I assumed it was something related to `HalfCauchy` being a special case of the above `Cauchy` distribution. The argument passed over here was a `TensorVariable` with a constant value of `{1}` during running of the `test_distribution`.

Reply:
Where do you find it requires `alpha`?

Is it possible it's coming from the new random method? The scipy method allows for loc and scale, but we only use scale in pymc3. We should make sure the alpha / beta is not being confused in the random calls, since beta is the second optional argument in the scipy version: https://github.com/pymc-devs/aesara/blob/b852bd24472e13ae2a405b36eaad462830c89228/aesara/tensor/random/basic.py#L228

Reply:
On running the test in `test_distribution` with only beta parameter, the `log_p` method got 4 parameters instead of 3. I assumed that had something to do with the mathematics involved. So I put in an extra argument. Strangely the value of `beta` is being passed first. The other one (fourth argument) always remains a constant.

Reply:
Still it seems that if a single variable is passed to the random op it will take it as a loc parameter, while the logp will take it as a scale parameter given the order of the arguments in the aesara op, no?

Reply:
Not to distract from the extra parameter, which is certainly a problem.